src/sparseml/pytorch/optim/modifier_pruning.py

```diff
@@ -212,48 +192,74 @@ def params(self, value: Union[str, List[str]]):
+        epoch: float = 0,
+        loggers: Optional[List[BaseLogger]] = None,
+        **kwargs,
+    ):
+        """
+        Grab the params and apply if epoch in range to control pruning for.
 
-        named_layers_and_params = get_named_layers_and_params_by_regex(
-            module,
-            param_names,
-            params_strict=True,
-        )
+        :param module: the PyTorch model/module to modify
+        :param epoch: The epoch to initialize the modifier and module at.
+            Defaults to 0 (start of the training process)
+        :param loggers: Optional list of loggers to log the modification process to
+        :param kwargs: Optional kwargs to support specific arguments
+            for individual modifiers.
+        """
+        super().initialize(module, epoch, loggers, **kwargs)
+        named_layers_and_params = self._create_named_layers_and_params(module)
+        layers = [nlp.layer for nlp in named_layers_and_params]
+        param_names = [nlp.param_name for nlp in named_layers_and_params]
+        layer_names = [nlp.layer_name for nlp in named_layers_and_params]
+        self._module_masks = self._create_pruning_mask(layers, layer_names, param_names)
+        self._analyzers = self._create_analyzers(layers, layer_names, param_names)
 
-        layers = [
-            named_layer_param.layer for named_layer_param in named_layers_and_params
-        ]
-        param_names = [
-            named_layer_param.param_name
-            for named_layer_param in named_layers_and_params
-        ]
-        layer_names = [
-            named_layer_param.layer_name
-            for named_layer_param in named_layers_and_params
-        ]
-        self._module_masks = ModuleParamPruningMask(
-            layers, param_names, layer_names=layer_names
-        )
+        if len(self._analyzers) == 0:
+            raise ValueError(
+                "Could not find any params matching {} in {}".format(
+                    self._params, self.__class__.__name__
+                )
+            )
 
-        self._analyzers = []
-        for layer_name, layer, param_name, _ in named_layers_and_params:
-            self._analyzers.append(ModulePruningAnalyzer(layer, layer_name, param_name))
+        self._check_mask_update(module, epoch, steps_per_epoch=1)
+
+    def finalize(
+        self, module: Optional[Module] = None, reset_loggers: bool = True, **kwargs
+    ):
+        """
+        Cleans up any remaining hooks
+
+        :param module: The model/module to finalize the modifier for.
+            Marked optional so state can still be cleaned up on delete,
+            but generally should always be passed in.
+        :param reset_loggers: True to remove any currently attached loggers (default),
+            False to keep the loggers attached.
+        :param kwargs: Optional kwargs to support specific arguments
+            for individual modifiers.
+        """
+        super().finalize(module, reset_loggers, **kwargs)
```

Comment:
let's call `apply()` one more time if the masks are enabled

Reply:
done